docs/api/proposed_endpoints.md

```diff
@@ -0,0 +1,153 @@
+[TOC]
+
+# Posts related actions
+
+## Create a post
+
+Create a post of contents (text + image).
+
+```
+POST api/posts/create
+```
+
+**Paramaters**
+
+Name         | Description
+-------------|--------------------------------------
+id           | identity of the post
+posted_by    | user who is posting the content
+posted_at    | social media ploatform to post
+posted_time  | time at which post should be posted
+content      | content of the post
+image        | image in the post
+is_approved  | whether the post is approved or not
+is_posted    | whether the post is posted or not
+
+**Request**
+```json
+{
+    "id": "724",
+    "posted_by": "John Micky",
+    "posted_at": "Twitter",
+    "posted_time": "",
+    "content": "Anything that user has to write",
+    "image": "Uploaded Image",
+    "is_approved": "False",
+    "is_posted": "False"
+}
+```
+
+**Response**
+```json
+
+Status: 201 Created
```

Comment:
This should ideally return the entire latest representation of the resource we have.